pkg/operators/localmanager/localmanager.go

```diff
@@ -422,6 +425,175 @@ func (l *localManagerTrace) EnrichEvent(ev any) error {
+			params:             l.ParamDescs().ToParams(), // TODO
+			gadgetCtx:          ctx,
+			eventWrappers:      make(map[datasource.DataSource]*compat.EventWrapperBase),
+		},
+	}
+
+	if l.igManager == nil {
+		traceInstance.enrichEvents = false
+	}
+
+	// hack - this makes it possible to use the Attacher interface
+	var ok bool
+	traceInstance.gadgetInstance, ok = ctx.GetVar("ebpfInstance")
+	if !ok {
+		return nil, fmt.Errorf("getting ebpfInstance")
+	}
+
+	activate := true // TODO: change to false once snapshot_socket is not skipped anymore
+
+	// Check, whether the gadget requested a map from us
+	if t, ok := ctx.GetVar(gadgets.MntNsFilterMapName); ok {
+		if _, ok := t.(*ebpf.Map); ok {
+			ctx.Logger().Debugf("gadget requested map %s", gadgets.MntNsFilterMapName)
+			activate = true
+		}
+	}
+
+	// Check for data sources containing refererences to mntns/netns that we could enrich data for
+	for _, ds := range ctx.GetDataSources() {
+		mntnsFields := ds.GetFieldsWithTag("type:gadget_mntns_id")               // TODO: const
+		netnsFields := ds.GetFieldsWithTag("type:gadget_netns_id", "name:netns") // TODO: const
+		if len(mntnsFields) == 0 && len(netnsFields) == 0 {
+			continue
+		}
+
+		ctx.Logger().Debugf("ds found: %q", ds.Name())
+
+		activate = true
+
+		var err error
+
+		var mntnsField datasource.FieldAccessor
+		var netnsField datasource.FieldAccessor
+
+		for _, f := range mntnsFields {
+			ctx.Logger().Debugf("using mntns enrichment")
+			mntnsField = f
+			mntnsField.SetHidden(true)
+			// We only support one of those per DataSource for now
+			break
+		}
+		for _, f := range netnsFields {
+			ctx.Logger().Debugf("using netns enrichment")
+			netnsField = f
+			netnsField.SetHidden(true)
+			// We only support one of those per DataSource for now
+			break
+		}
+
+		traceInstance.eventWrappers[ds], err = compat.WrapAccessors(ds, mntnsField, netnsField)
+		if err != nil {
+			return nil, fmt.Errorf("registering accessors: %w", err)
+		}
+	}
+
+	if !activate {
+		return nil, nil
+	}
+
+	return traceInstance, nil
+}
+
+func (l *localManagerTrace) ParamDescs() params.ParamDescs {
+	return params.ParamDescs{
+		{
+			Key:         ContainerName,
+			Alias:       "c",
+			Description: "Show only data from containers with that name",
+			ValueHint:   gadgets.LocalContainer,
+		},
+		{
+			Key:          Host,
+			Description:  "Show data from both the host and containers",
+			DefaultValue: "false",
+			TypeHint:     params.TypeBool,
+		},
+	}
+}
+
+func (l *localManagerTraceWrapper) ParamDescs(gadgetCtx operators.GadgetContext) params.ParamDescs {
+	return l.localManagerTrace.ParamDescs()
+}
+
+func (l *localManagerTraceWrapper) Prepare(gadgetCtx operators.GadgetContext, params *params.Params) error {
+	l.params = params
+	for ds, wrapper := range l.eventWrappers {
+		ds.Subscribe(func(ds datasource.DataSource, data datasource.Data) error {
+			wr := &compat.EventWrapper{
```

Comment:
In this case, we don't - but the func to Subscribe has this signature to also be used in different contexts, where you might not have the `ds` from the loop like here.

Example: You implement a new struct with a method on it that has the same signature (`func (s *myNewStruct) MyMethod(ds datasource.DataSource, data datasource.Data) error { ... }` - then you can just call `ds.Subscribe(&myNewStruct{}.MyMethod)` without needing to store the datasource somewhere - especially if you apply the same method for many different (but applicable) DataSources, it can be really handy to have the additional reference.

Reply:
> > In this case, we don't - but the func to Subscribe has this signature to also be used in different contexts, where you might not have the `ds` from the loop like here.
> 
> Is this really needed? This PR refactors and the code and I except it to remove more code than it should add (once we would drop the existing code). So, I would rather focus on pragmatic and existing use case rather than adding code "just in case".

Needed here, as said: no. But it's part of our API - every operator in the future will use this code to receive data from DataSources. I've run into similar situations where the context (in this case the DataSource) was missing from a signature and I had to add complex ways (including mem allocations) to store the context, just to be able to properly handle things.

> 
> > Example: You implement a new struct with a method on it that has the same signature (`func (s *myNewStruct) MyMethod(ds datasource.DataSource, data datasource.Data) error { ... }` - then you can just call `ds.Subscribe(&myNewStruct{}.MyMethod)` without needing to store the datasource somewhere - especially if you apply the same method for many different (but applicable) DataSources, it can be really handy to have the additional reference.
> 
> Can you please give me a real world example of this?

Phew, thanks for the challenge 😄 Gave it a couple minutes now but I can't really come up with an _optimized_ example (optimized meaning that you would at least cache one accessor somehow, with which you could then also cache the DataSource). Will think further about an example, but I still think it's worth adding that pointer "just in case" to such an API 😄 

BUT: this is only due to the most recent Go version - before that, looping over pointers like that would have made the outer `ds` value always contain the last value of `l.eventWrappers` when accessed in the callback function, so you'd have to use something like `tempDS := ds` inside the loop to have a proper reference for the closure.